source/data-api/custom-endpoints.txt

```diff
@@ -0,0 +1,676 @@
+
+         {
+           "run_as_user_id": "628e47baf4c2ac2796fc8a91"
+         }
+   
+   .. tab:: Script
+      :tabid: auth-script
+   
+      Script authentication calls a function to determine which
+      application user a request runs as.
+
+      The function must return an existing application user's Account ID
+      as a string or ``{ "runAsSystemUser": true }`` to run the request
+      as a :ref:`system user <system-user>` that has full access to
+      MongoDB CRUD and Aggregation APIs and is not subject to any
+      rules, roles, or limited permissions.
+      
+      .. code-block:: json
+         :caption: http_endpoints/config.json
+
+         {
+           "run_as_user_id_script_source": "exports = () => {return \"628e47baf4c2ac2796fc8a91\"}"
+         }
+   
+   .. tab:: System
+      :tabid: auth-system
+
+      System authentication configures an endpoint to run as a
+      :ref:`system user <system-user>` that requires no credentials, has
+      full access to MongoDB CRUD and Aggregation APIs, and is not
+      subject to any rules, roles, or limited permissions.
+      
+      .. code-block:: json
+         :caption: http_endpoints/config.json
+
+         {
+           "run_as_system": true
+         }
+
+.. tip::
+   
+   New endpoints that you create in the UI use System authentication by default.
+
+.. _endpoint-authorization:
+
+Authorization
+~~~~~~~~~~~~~
+
+An endpoint can require authenticated users to provide additional
+authorization information in the request. You define the authorization
+scheme for each custom endpoint by configuring the endpoint function.
+
+.. tabs::
+     
+   .. tab:: No Additional Authorization
+      :tabid: no-authorization
+   
+      All authenticated users are authorized to call the endpoint.
+      Authenticated requests do not need to include authorization
+      information.
+      
+   .. tab:: Secret Parameter
+      :tabid: secret-parameter
+      
+      Authenticated users must prove that they are authorized to call
+      the endpoint by including a secret value as the value of the
+      ``secret`` query parameter.
+
+      .. example::
+         
+         The following ``curl`` request uses secret query parameter
+         validation with the secret value ``12345``:
+      
+         .. code-block:: bash
+            :emphasize-lines: 4
+      
+            curl -X POST \
+                 -H "Content-Type: application/json" \
+                 -d '{ "message": "HELLO" }' \
+                 https://data.mongodb-api.com/app/myapp-abcde/endpoint/sendMessage?secret=12345
+   
+   .. tab:: Payload Signature
+      :tabid: payload-signature
+   
+      Authenticated users must prove that they are authorized to call
+      the endpoint by including an ``Endpoint-Signature`` header that
+      contains a hexadecimal-encoded :abbr:`HMAC (Hash-based Message
+      Authentication Code)` SHA-256 hash generated from the request body
+      and the secret value.
```

Comment:
i find this section insufficient info, esp how it relates to endpoints using secrets auth. could you expand?

Reply:
Ok added some info about defining a secret (I actually didn't realize they changed this to reference a secret by name instead of defining the string directly)